include/dem/dem.h

```diff
@@ -201,8 +210,15 @@ class DEMSolver
   TimerOutput                               computing_timer;
   Particles::ParticleHandler<dim, dim>      particle_handler;
   int                                       DEM_step = 1;
-  double       DEM_time = parameters.simulationControl.dt;
-  const double dt       = parameters.simulationControl.dt;
+  double       DEM_time           = parameters.simulationControl.dt;
+  const double dt                 = parameters.simulationControl.dt;
+  unsigned int inserted_this_step = 0;
+  unsigned int remained_particles =
+    parameters.simulationControl.total_particle_number;
+  bool         insertion_step = 0;
+  unsigned int number_of_particles_x_direction, number_of_particles_y_direction,
+    number_of_particles_z_direction;
+
```

Comment:
I don't think these parameters should be stored by the DEM class. It should be the insertion class that knows these elements. Futher more, the number of particles in x , y and z direction are already stored in the parameters::insertion, why are we storing a second copy of them?